src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs

```diff
@@ -10,21 +10,38 @@
 using UrbanAirship.MessageCenter;
 
 namespace UrbanAirship.NETStandard
-{
+{ 
     public delegate void DeepLinkHandler(string deepLink);
 
-    public class Airship : Java.Lang.Object, IDeepLinkListener, IAirship
+    public class Airship : Java.Lang.Object, IDeepLinkListener, IAirship, UrbanAirship.Channel.IAirshipChannelListener
     {
-        private static Airship sharedAirship = new Airship();
+
+        public delegate void ChannelHandler(string channelId);
+
+        private static Lazy<Airship> sharedAirship = new Lazy<Airship>(() =>
+        {
+            Airship instance = new Airship();
+            instance.Init();
+            return instance;
+        });
+
+        private void Init()
+        {
+            UAirship.Shared().Channel.AddChannelListener(this);
+        }
 
         public static Airship Instance
         {
             get
             {
-                return sharedAirship;
+                return sharedAirship.Value;
             }
         }
 
+        public event ChannelHandler onChannelCreation;
```

Comment:
instead of using a custom delegate, I'd prefer that we use EventHandler here. Make a class `ChannelEventArgs` that extends EventArgs, and then:

`public event EventHandler<ChannelEventArgs> onChannelCreation`

etc

https://docs.microsoft.com/en-us/dotnet/api/system.eventhandler-1?view=netcore-3.1

Reply:
We should also make sure that these new events are in IAirship.cs, and using EventHandler<T> will make that easier since that type is already defined externally.

Reply:
Can you look if it's ok @marc-scig pls

Reply:
Okay I push an update but I get this: 'Airship' does not implement interface member 'IAirship.OnChannelCreation' and 'Airship' does not implement interface member 'IAirship.OnChannelUpdate'

Reply:
@benjamin-benoit that's probably because there's a type mismatch. I'll point it out in the diff